fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/PartitionColPredicateEvaluator.java

```diff
@@ -0,0 +1,197 @@
+                    break;
+                case LT:
+                    predicateRange = Range.lessThan(conditionKey);
+                    break;
+                case NE:
+                default:
+                    predicateRange = Range.all();
+            }
+            return evaluateRangeHitSet(predicateRange);
+        }
+
+        @Override
+        public BitSet visitCompoundPredicate(CompoundPredicateOperator predicate, Void context) {
+            BitSet first = predicate.getChild(0).accept(this, null);
+            if (predicate.isAnd()) {
+                BitSet second = predicate.getChild(1).accept(this, null);
+                first.and(second);
+            } else if (predicate.isOr()) {
+                BitSet second = predicate.getChild(1).accept(this, null);
+                first.or(second);
+            } else {
+                first = predicate.getChild(0).accept(this, null);
+                first.flip(0, candidateNum);
+            }
+            return first;
+        }
+
+        // only process in scene
+        @Override
+        public BitSet visitInPredicate(InPredicateOperator predicate, Void context) {
+            List<ConstantOperator> constList = predicate.getChildren().stream()
+                    .skip(1).map(e -> (ConstantOperator) e)
+                    .filter(e -> !e.isNull()).sorted().collect(Collectors.toList());
+            BitSet res = new BitSet();
+            boolean encounterEx = false;
+            if (IN_OPERANDS_LIMIT >= constList.size()) {
+                for (ConstantOperator constantOperator : constList) {
+                    try {
+                        LiteralExpr literalExpr = ColumnFilterConverter.convertLiteral(constantOperator);
+                        PartitionKey conditionKey = new PartitionKey();
+                        conditionKey.pushColumn(literalExpr, partitionColumn.getPrimitiveType());
+                        Range<PartitionKey> predicateRange = Range.closed(conditionKey, conditionKey);
+                        res.or(evaluateRangeHitSet(predicateRange));
+                    } catch (AnalysisException e) {
+                        encounterEx = true;
+                        break;
+                    }
+                }
+            } else {
+                try {
+                    LiteralExpr min = ColumnFilterConverter.convertLiteral(constList.get(0));
+                    LiteralExpr max = ColumnFilterConverter.convertLiteral(constList.get(constList.size() - 1));
+                    PartitionKey minKey = new PartitionKey();
+                    minKey.pushColumn(min, partitionColumn.getPrimitiveType());
+                    PartitionKey maxKey = new PartitionKey();
+                    maxKey.pushColumn(max, partitionColumn.getPrimitiveType());
+                    Range<PartitionKey> coarseRange = Range.range(minKey, BoundType.CLOSED, maxKey, BoundType.CLOSED);
+                    res.or(evaluateRangeHitSet(coarseRange));
```

Comment:
In this way, you can accurately find the partition? This still looks like using range to find partition